File: src/main/resources/static/js/app/AtmService.js
'use strict'

angular.module('demo.services').factory('AtmService',
    [ "$http", "CONSTANTS", function($http, CONSTANTS) {
        var service = {};
        service.getAtmById = function(atmId) {
            var url = CONSTANTS.getAtmByIdUrl + atmId;
            return $http.get(url);
        }
        service.getAllAtms = function() {
            return $http.get(CONSTANTS.getAllAtms);
        }
        service.saveAtm = function(atmDto) {
            return $http.post(CONSTANTS.saveAtm, atmDto);
        }
        service.withdrawBills = function (atmDto, withdrawal) {
            var url = CONSTANTS.withdrawBillsUrl + withdrawal;
            return $http.post(url, atmDto);
        }
        service.depositBills = function (atmDto, deposit20bill, deposit50bill, deposit100bill, deposit200bill, deposit500bill, deposit1000bill) {
            var url = CONSTANTS.depositBillsUrl + deposit20bill + "/" + deposit50bill +"/" + deposit100bill + "/" + deposit200bill + "/" + deposit500bill + "/" + deposit1000bill;
            return $http.post(url, atmDto);
        }
        return service;
    } ]);

File: src/main/java/com/example/simpleatm/service/AtmService.java
package com.example.simpleatm.service;

import com.example.simpleatm.dto.AtmDto;
import java.util.List;

/**
 * Created by vgerris on 29/4/19.
 */
public interface AtmService {
    AtmDto getAtmById(Integer atmId);
    void saveAtm(AtmDto atmDto);
    Float withdrawBills(AtmDto atmDto, Float withdrawal);
    Float depositBills(AtmDto atmDto, Float deposit20, Float deposit50, Float deposit100, Float deposit200, Float deposit500, Float deposit1000);
    List<AtmDto> getAllAtms();
}